### ISSUE TITLE
bats: Check for negation that won't fail the test

### ISSUE THREAD

**Issue by martin-schulze-vireso** [koalaman/shellcheck#2273]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

See https://github.com/bats-core/bats-core/issues/462 for more context.
In short: Using negation on a statement (that is not the last) does not trigger a test failure for bats because bash does not trigger the error trap on that. Since this is a deliberate bash decision, we need to educate users about it.

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env bats

@test "name" {
    ! true # this line should fail the test but doesn't
    true
}
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Warning on the line `! true`: "Negation won't trigger test failure due to bash's design decision not to trigger the error trap on negated statements."
